### ISSUE TITLE
Sharing big directory hits execution timeout and fails

### ISSUE THREAD

**Issue by sargue** [owncloud/core#12734]
I posted [a bug](https://github.com/owncloud/client/issues/2566) to the mirall client which seems to be related to the server. In a nutshell, sharing a big directory between users on an encrypted owncloud might hit execution timeout and silently fail. You get this on the logs:

```
{"app":"PHP","message":"Maximum execution time of 3600 seconds exceeded at \/var\/local\/www-hosts\/domain.name\/owncloud\/lib\/private\/hook.php#36","level":3,"time":"2014-11-28T12:29:57+00:00"}
{"app":"OCP\\Share","message":"Sharing libros failed, because this item is already shared with bailams","level":3,"time":"2014-11-28T12:30:00+00:00"} 
```

See the [original bug](https://github.com/owncloud/client/issues/2566) for more details. Maybe sharing may be delegated to a background task on the scheduler (cron,ajax...) and be done on chunks? The issue #10657 seems somehow related.

**Comment by PVince81** [comment]
Duplicate of https://github.com/owncloud/core/issues/12413
